### PR TITLE
ibm5170: new software list addition

### PR DESCRIPTION
**Pull request by davide125**
Converted to IMD with `disk-analyze` from the SCP dumps at https://archive.org/details/Doom1BBS (as I don't think MAME supports SCP directly). IMD files are at https://drive.google.com/drive/folders/1thWkdG0ZEtyW19GI7XLpm_w98J5kXtGY?usp=sharing

### CODE REVIEW

**Comment by DopefishJustin** [comment]
The softlists are supposed to be long-term preservation so we have been trying to avoid formats like IMD that are proprietary and undocumented, even if they are supported to a greater or lesser degree. For the ibm5170 softlist we have been using either a plain sector image (.img or .dsk) if there is no copy protection or special on-disk format, or the raw .mfm format from HxCFloppyEmulator if there is something special. I wouldn't expect there to be any protection on a BBS disk like this though.

**Comment by davide125** [comment]
I used IMD because the SCP dumps contain some unformatted tracks, and I couldn't get a valid IMG sector image out of it.

**Comment by einstein95** [comment]
Converting to IMG and specifying to use `ibm` format worked fine. [Here's the resulting IMG files](https://github.com/mamedev/mame/files/5253580/Doom_BBS.zip).
```
$ disk-analyse -f ibm Doom_BBS_1.scp Doom_BBS_1.img
T0.0-79.1: IBM-MFM HD (18 512-byte sectors, 9216 bytes)
T80.0-83.1: Unformatted

$ disk-analyse -f ibm Doom_BBS_2.scp Doom_BBS_2.img
T0.0-68.0: IBM-MFM HD (18 512-byte sectors, 9216 bytes)
T68.1-83.1: Unformatted
```
Change the XML entries to
```xml
<dataarea name="flop" size="1474560">
	<rom name="Doom_BBS_1.img" size="1474560" crc="e9b64e07" sha1="2f2a7ddc19b1c68696deaa06296684a6b0f4c538"/>
</dataarea>


<dataarea name="flop" size="1262592">
	<rom name="Doom_BBS_2.img" size="1262592" crc="8df21831" sha1="50c37cf3b36f4ddfeb315496a4e288d3e64a710a"/>
</dataarea>
```

**Comment by davide125** [comment]
Thanks! I'll update the PR later tonight.

**Comment by davide125** [comment]
Updated, thanks!

File: hash/ibm5170.xml [hunk 1]
```diff
@@ -10355,6 +10355,29 @@ license:CC0
...
+			<feature name="part_id" value="Disk 1 of 2"/>
+			<dataarea name="flop" size="1474050">
+				<rom name="Doom_BBS_1.img" size="1474560" crc="e9b64e07" sha1="2f2a7ddc19b1c68696deaa06296684a6b0f4c538"/>
+			</dataarea>
+		</part>
+		<part name="flop2" interface="floppy_3_5">
+			<feature name="part_number" value="0863-2"/>
+			<feature name="part_id" value="Disk 2 of 2"/>
+			<dataarea name="flop" size="816058">
+				<rom name="Doom_BBS_2.img" size="1262592" crc="8df21831" sha1="50c37cf3b36f4ddfeb315496a4e288d3e64a710a"/>
```
Comment (DopefishJustin) [inline hash/ibm5170.xml:10376]: 1) dataarea size doesn't match rom size
2) File size should be 1474560 for a 1.44Mb disk. In other cases I have padded with 0xF6 which is the default for unused sectors on PC but it doesn't really matter what byte is used.

Comment (davide125) [reply hash/ibm5170.xml]: Thanks! PR updated, and I've uploaded the padded image at https://drive.google.com/drive/folders/1thWkdG0ZEtyW19GI7XLpm_w98J5kXtGY?usp=sharing

**Pull request merged**